api/host.go

```diff
@@ -166,40 +167,109 @@ func (api *API) hostHandlerPOST(w http.ResponseWriter, req *http.Request, _ http
 		var x types.Currency
 		_, err := fmt.Sscan(req.FormValue("mincontractprice"), &x)
 		if err != nil {
-			WriteError(w, Error{"Malformed mincontractprice"}, http.StatusBadRequest)
-			return
+			return modules.HostInternalSettings{}, nil
 		}
 		settings.MinContractPrice = x
 	}
 	if req.FormValue("mindownloadbandwidthprice") != "" {
 		var x types.Currency
 		_, err := fmt.Sscan(req.FormValue("mindownloadbandwidthprice"), &x)
 		if err != nil {
-			WriteError(w, Error{"Malformed mindownloadbandwidthprice"}, http.StatusBadRequest)
-			return
+			return modules.HostInternalSettings{}, nil
 		}
 		settings.MinDownloadBandwidthPrice = x
 	}
 	if req.FormValue("minstorageprice") != "" {
 		var x types.Currency
 		_, err := fmt.Sscan(req.FormValue("minstorageprice"), &x)
 		if err != nil {
-			WriteError(w, Error{"Malformed minstorageprice"}, http.StatusBadRequest)
-			return
+			return modules.HostInternalSettings{}, nil
 		}
 		settings.MinStoragePrice = x
 	}
 	if req.FormValue("minuploadbandwidthprice") != "" {
 		var x types.Currency
 		_, err := fmt.Sscan(req.FormValue("minuploadbandwidthprice"), &x)
 		if err != nil {
-			WriteError(w, Error{"Malformed minuploadbandwidthprice"}, http.StatusBadRequest)
-			return
+			return modules.HostInternalSettings{}, nil
 		}
 		settings.MinUploadBandwidthPrice = x
 	}
 
-	err := api.host.SetInternalSettings(settings)
+	return settings, nil
+}
+
+// hostEstimateScoreGET handles the POST request to /host/estimatescore and
+// computes an estimated HostDB score for the provided settings.
+func (api *API) hostEstimateScorePOST(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	entry, exists := api.renter.Host(api.host.PublicKey())
```

Comment:
We overwrite the settings with the previous settings combined with the settings provided to the endpoint, but the uptime is left unchanged. This is so the estimate takes into account your current uptime.

Come to think of it, that does actually pose a problem. We learned recently that certain network configurations do not allow you to probe your own external IP address, meaning that users with that problem would have an unfairly low estimation of their score.

Not sure what the best way of dealing with that is, we could simply ignore uptime and not fetch the entry.